torch/csrc/functorch/init.cpp

```diff
@@ -282,8 +298,8 @@ static Tensor get_unwrapped(const Tensor& tensor) {
   if (wrapped) {
     return wrapped->value();
   }
-  auto* functional = dynamic_cast<FunctionalTensorWrapper*>(tensor.unsafeGetTensorImpl());
-  if (functional) {
+  if (at::functionalization::impl::isFunctionalTensor(tensor)) {
+    auto* functional = at::functionalization::impl::unsafeGetFunctionalWrapper(tensor);
```

Comment:
dynamic_cast isn't portable (due to needing rtti) so I changed how we coerce a Tensor into a FunctionalTensorWrapper